onnx/defs/generator/defs.cc

```diff
@@ -1,227 +1,282 @@
+    .TypeConstraint("T2", { "tensor(float16)", "tensor(float)", "tensor(double)" },
+        "Constrain output types to float tensors.")
+    .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+        if (ctx.getAttribute("dtype") != nullptr) 
+          propagateElemTypeFromAttributeToOutput(ctx, "dtype", 0);
+        else
+          propagateElemTypeFromInputToOutput(ctx, 0, 0);
+        if (!hasNInputShapes(ctx, 1)) {
+          return;
+        }
+        propagateShapeFromInputToOutput(ctx, 0, 0);
+    });
+
+ONNX_OPERATOR_SCHEMA(RandomNormalLike)
+    .SetDoc(R"DOC(
+Generate a tensor with random values drawn from a normal distribution. 
+The shape of the output tensor is copied from the shape of the input tensor, 
+and the parameters of the normal distribution are specified by `mean` and `scale`.
+
+The data type is specified by the 'dtype' argument, or copied from the input tensor if not provided. 
+The 'dtype' argument must be one of the data types specified in the 'DataType' enum field in the
+TensorProto message, and be valid as an output type.
+)DOC")
+    .Attr(
+          "mean",
+          "The mean of the normal distribution. If not specified, default is 0.",
+          AttributeProto::FLOAT,
+          0.0f)
+    .Attr(
+          "scale",
+          "The standard deviation of the normal distribution. If not specified, default is 1.",
+          AttributeProto::FLOAT,
+          1.0f)
+    .Attr(
+          "seed",
+          "(Optional) Seed to the random generator, if not specified we will auto generate one.",
+          AttributeProto::FLOAT,
+          OPTIONAL)
+    .Attr(
+          "dtype",
+          "(Optional) The data type for the elements of the output tensor, if not specified, we will use"
+          "the data type of the input tensor.",
+          AttributeProto::INT,
+          OPTIONAL)
+    .Input(
+           0,
+           "input",
+           "Input tensor to copy shape and optionally type information from.", "T1")
+    .Output(
+            0,
+            "output",
+            "Output tensor of random values drawn from normal distribution", "T2")
+    .TypeConstraint(
+        "T1",
+        OpSchema::all_tensor_types(),
+        "Constrain to any tensor type. If the dtype attribute is not provided this must be a valid output type.")
+    .TypeConstraint("T2", { "tensor(float16)", "tensor(float)", "tensor(double)" },
+        "Constrain output types to float tensors.")
+    .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+        if (ctx.getAttribute("dtype") != nullptr) 
+          propagateElemTypeFromAttributeToOutput(ctx, "dtype", 0);
+        else
+          propagateElemTypeFromInputToOutput(ctx, 0, 0);
+        if (!hasNInputShapes(ctx, 1)) {
+          return;
+        }
+        propagateShapeFromInputToOutput(ctx, 0, 0);
+    });
+
+ONNX_OPERATOR_SCHEMA(Multinomial)

```

Comment:
why isn't this experimental?

Reply:
Why shouldn't it be real if there is a clear use case for it in a model? This operator is requested by Unity. @prasanthpul can you help facilitate this conversation?

Reply:
@anirudhacharya generally ops are added as experimental because their use or value is uncertain. for example it could be an op that is only used in 1 framework or an op that potentially could be decomposed into other ops without a perf hit. In this case, the op is supported by multiple frameworks such as TF, PyTorch, mxnet, etc. and can be considered a primitive. So the proper course is to add it as a core op - similar to how the trig ops were added recently.